staging/src/k8s.io/kubectl/pkg/cmd/set/helper.go

```diff
@@ -138,6 +138,8 @@ func updateEnv(existing []v1.EnvVar, env []v1.EnvVar, remove []string) []v1.EnvV
 		if covered.Has(e.Name) {
 			continue
 		}
+		// If it is an added Env, the duplicate Env will be filtered and the value of the newly added env will be retained.
+		// If not, it remains unchanged.
```

Comment:
Adding documentation about how the update works sounds good to me. Since this doesn't describe the code directly below in particular, let's move it up to be a function comment (directly above `updateEnv`) and broaden it a bit to describe what the function does overall.